src/Sarif.Viewer.VisualStudio.Shell.Core/InfoBarService.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+using CSharpFunctionalExtensions;
+
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Sarif.Viewer.Shell
+{
+    /// <inheritdoc cref="IInfoBarService"/>
+    public class InfoBarService : IInfoBarService, IVsInfoBarUIEvents
+    {
+        private readonly IServiceProvider serviceProvider;
+        private uint cookie;
+
+        public InfoBarService(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+        }
+
+        /// <inheritdoc cref="IInfoBarService.ShowInfoBar(InfoBarModel)"/>
+        public IVsInfoBarUIElement ShowInfoBar(InfoBarModel infoBarModel)
```

Comment:
>IVsInfoBarUIElement [](http://example.com/codeflow?start=15&length=19)

why not return Result<...> as other methods do?

Reply:
Yes should be Maybe<> in this case.

Reply:
Ah I remember why I could,'t do it this way. IVsInfoBarUIElement can't be passed between assemblies as a generic type.
[https://stackoverflow.com/questions/35270785/c-sharp-cannot-be-used-across-assembly-boundaries-because-it-has-a-generic-type](https://stackoverflow.com/questions/35270785/c-sharp-cannot-be-used-across-assembly-boundaries-because-it-has-a-generic-type)